src/desktop/apps/authentication/helpers.ts

```diff
@@ -12,6 +12,41 @@ import { captureException } from "@sentry/browser"
 const mediator = require("../../lib/mediator.coffee")
 const LoggedOutUser = require("../../models/logged_out_user.coffee")
 
+/**
+ * Open authentication modal via 'click' trigger
+ */
+export const handleOpenAuthModal = (mode: ModalType, options: ModalOptions) => {
+  mediator.trigger("open:auth", {
+    mode,
+    ...options,
+  })
+}
+
+/**
+ * Set up scroll event to open authentication modal via 'timed' trigger
+ * Opens 2 seconds after first scroll by default
+ */
+export const handleScrollingAuthModal = (options: ModalOptions) => {
+  if (!sd.CURRENT_USER && !sd.IS_MOBILE) {
```

Comment:
#minor nit - its generally preferred to return early vs gating the entire body of a function